QgisModelBaker/libqgsprojectgen/dbconnector/mssql_connector.py

```diff
@@ -0,0 +1,421 @@
+                    {model_name}
+                    {ili_name}
+                    {extent}
+                    tgeomtype.setting as simple_type,
+                    null as formatted_type
+                FROM
+                INFORMATION_SCHEMA.TABLE_CONSTRAINTS Tab 
+                INNER JOIN INFORMATION_SCHEMA.CONSTRAINT_COLUMN_USAGE Col 
+                    on Col.Constraint_Name = Tab.Constraint_Name
+                    AND Col.Table_Name = Tab.Table_Name
+                    AND Col.CONSTRAINT_SCHEMA = Tab.CONSTRAINT_SCHEMA
+                RIGHT JOIN INFORMATION_SCHEMA.TABLES as tbls
+                    on Tab.TABLE_NAME = tbls.TABLE_NAME 
+                    AND Tab.CONSTRAINT_SCHEMA = tbls.TABLE_SCHEMA
+                    AND Tab.Constraint_Type = 'PRIMARY KEY'
+                {domain_left_join}
+                {alias_left_join}
+                {model_where}
+                LEFT JOIN INFORMATION_SCHEMA.COLUMNS as clm
+                    on clm.TABLE_NAME = tbls.TABLE_NAME
+                    AND clm.TABLE_SCHEMA = tbls.TABLE_SCHEMA
+                    AND clm.DATA_TYPE = 'geometry'
+                LEFT JOIN {schema}.T_ILI2DB_COLUMN_PROP as tsrid
+                    on tbls.TABLE_NAME = tsrid.tablename and clm.COLUMN_NAME = tsrid.columnname and tsrid.tag='ch.ehi.ili2db.srid'
+                LEFT JOIN {schema}.T_ILI2DB_COLUMN_PROP as tgeomtype
+                    on tbls.TABLE_NAME = tgeomtype.tablename and clm.COLUMN_NAME = tgeomtype.columnname and tgeomtype.tag= 'ch.ehi.ili2db.geomType' 
+                WHERE tbls.TABLE_TYPE = 'BASE TABLE' {schema_where}
+            """.format(kind_settings_field=kind_settings_field, table_alias=table_alias, model_name=model_name, ili_name=ili_name, extent=extent,
+                    domain_left_join=domain_left_join, alias_left_join=alias_left_join, model_where=model_where,
+                    schema_where=schema_where,schema=self.schema)
+            cur.execute(query)
+
+            columns = [column[0] for column in cur.description]
+
+            for row in cur.fetchall():
+                my_rec = dict(zip(columns, row))
+                # TODO type != simple_type
+                my_rec['type'] = my_rec['simple_type']
+                res.append(my_rec)
+
+        return res
+
+    def get_meta_attrs(self, ili_name):
+        if not self._table_exists(METAATTRS_TABLE):
+            return []
+        
+        result = []
+        
+        if self.schema:
+            cur = self.conn.cursor()
+            cur.execute("""
+                        SELECT
+                          attr_name,
+                          attr_value
+                        FROM {schema}.{metaattrs_table}
+                        WHERE ilielement='{ili_name}';
+            """.format(schema=self.schema, metaattrs_table=METAATTRS_TABLE, ili_name=ili_name))
+
+            result = self._get_dict_result(cur)
+
+        return result
+
+    def get_fields_info(self, table_name):
```

Comment:
same remark with code duplication for this method

Reply:
Same response, SQLs are specific for each DB engine.